### ISSUE TITLE
Complex system command output is suppressed when jruby is executed from GlassFish.

### ISSUE THREAD

**Issue by annabackiyam** [jruby/jruby#5833]
JRuby version  - 9.1.13 (-Djruby.native.enabled=false)
Operating system and platform - Linux and GlassFish

**Actual Problem:**
The output is empty when the following script is run from GlassFish environment

#!/usr/local/bin/ruby
IO.popen(’( dbus-uuidgen > /var/lib/dbus/machine-id ; /usr/bin/Xvnc -rfbauth /etc/hosts & export DISPLAY=:0 ; ( /usr/bin/firefox & ) > /dev/null 2>&1 ; sleep 5 ; /usr/bin/killall firefox ; /usr/bin/killall Xvnc ; if [ -d /root/.mozilla ]; then echo “SUCCESS” ; else echo “FAILED” ; fi)’) {
|f| puts f.gets
}

**Expected Output:**
Xvnc TigerVNC 1.1.0 - built Mar 28 2016 04:48:20
Copyright (C) 1999-2011 TigerVNC Team and many others (see README.txt)
See http://www.tigervnc.org for information on TigerVNC.
Underlying X server release 11704000, The X.Org Foundation

Tue Aug 13 21:53:21 2019
 vncext:      VNC extension running!
 vncext:      Listening for VNC connections on all interface(s), port 5900
 vncext:      created VNC server for screen 0
SUCCESS

**Comment by kares** [comment]
that seems like a standard embedded use-case.
maybe you could try reproducing setting up a `ScriptingContainer` and run the script, as a reproducer.
also did you try latest JRuby? even 9.1 had 4 releases (9.1.17) but all work is going towards 9.2 atm.

**Comment by annabackiyam** [comment]
@kares we are using Bean Scripting Framework (BSF)

**Comment by headius** [comment]
Hello again @annabackiyam! Is this a regression in a recent JRuby version or is this how it has always behaved?

I have not looked at BSF in a *very* long time, but I think like most other embedding frameworks that stdio will be shunted to in-memory streams, rather than just being dumped to the console. You should be able to configure it to use actual stdio, I think?

**Comment by annabackiyam** [comment]
Hello @headius, the issue started only from 9.1.13. It works in the earlier version.
I do have -Djruby.native.enabled=false. That helped for simple commands and not for the complicated one.

**Comment by headius** [comment]
@annabackiyam Are you able to bisect when it started happening?

**Comment by annabackiyam** [comment]
@headius Yes, I have tried that too. The commands are executed fine, just that output is empty in all cases.
